### ISSUE TITLE
Network offerings not listed on zone with multiple physical networks having guest traffic

### ISSUE THREAD

**Issue by rajujith** [apache/cloudstack#9673]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI, API
~~~

##### CLOUDSTACK VERSION

~~~
4.19.1.1
~~~

##### CONFIGURATION

Advanced zone with 4 physical networks
Public
Management
Guest (1) - VXLAN isolation, tag: vxlan
Guest (2) - VLAN Isolation, tag: vlan

##### OS / ENVIRONMENT

ACS 4.19.1.1 , KVM 

##### SUMMARY

After creating a zone with the above configuration I could create an isolated guest network on Guest (1), and a shared guest network on Guest (2).  But the network offering other than the 'quick cloud no service' no other network offering is listed. This issue remains even after removing the second physical network with the guest traffic. 

The listing works fine after removing the tag on the physical networks. Tagging existing physical network with guest is mandatory to add another physical network and add a traffic type guest to it. 

##### STEPS TO REPRODUCE

~~~
1. Create an advanced zone with 4 physical networks having traffic types Public, Management, Guest , Guest. 
2. One of the Guest can be VXLAN, other can be VLAN although not sure if it is causing the issue.
3. Tag the physical networks having guest traffic.
4. Create a network offering with a tag matching one of the physical network having guest traffic. 
5. Create a guest network. 
~~~

##### EXPECTED RESULTS

~~~
Network offerings get listed.
~~~

##### ACTUAL RESULTS

~~~
Network offerings do not get listed. 
~~~
![Screenshot 2024-09-12 at 12 17 34 PM](https://github.com/user-attachments/assets/bd0f10d2-d47b-47e1-8882-9073e5c30b26)

![multi guest](https://github.com/user-attachments/assets/c206f2ac-230a-405c-9b55-7558ca0f6d08)

**Comment by weizhouapache** [comment]
@rajujith 
you need to remove the tag of one guest physical network. or create network offerings with specific tag (vlan or vxlan)

**Comment by weizhouapache** [comment]
> good point [@weizhouapache](https://github.com/weizhouapache) , only one physical guest network without tags may exist as I recall. So also offerings without tags are only valid for that physnet, isn't that the case?
> 

yes, at most 1.

if users add 2nd physical network, they just need to tag the 2nd physical network. The existing network offerings (untagged) will keep working with the 1st physical network (untagged). This helps users when they add new guest physical network.
However, if users tag the 1st physical network as well, the existing network offering (untagged) will not work as all physical networks are tagged. then they have to create network offerings with the tags.


> Sounds like we need to revisit the functional spec for this one. cc [@Pearl1594](https://github.com/Pearl1594) [@rajujith](https://github.com/rajujith)

**Comment by rajujith** [comment]
@weizhouapache  If we are assuming the zone has only two physical networks with traffic type guest tagging one physical network may work, although I haven't tested it. What about a zone with 3 physical networks having traffic type of type guest. When I tried adding the 3rd Physical network with traffic type there was an error thrown that there is at least one physical network present in the zone having traffic type guest but not having a tag. I could add the guest traffic type to the third physical network only after adding tag to the second physical network as well. CC @Pearl1594 @DaanHoogland

**Comment by DaanHoogland** [comment]
> [@weizhouapache](https://github.com/weizhouapache) If we are assuming the zone has only two physical networks with traffic type guest tagging one physical network may work, although I haven't tested it. What about a zone with 3 physical networks having traffic type of type guest. When I tried adding the 3rd Physical network with traffic type there was an error thrown that there is at least one physical network present in the zone having traffic type guest but not having a tag. I could add the guest traffic type to the third physical network only after adding tag to the second physical network as well. CC [@Pearl1594](https://github.com/Pearl1594) [@DaanHoogland](https://github.com/DaanHoogland)

Yes @rajujith , there can be only one! (phys network for traffictype guest with no tag)

**Comment by weizhouapache** [comment]
> > [@weizhouapache](https://github.com/weizhouapache) If we are assuming the zone has only two physical networks with traffic type guest tagging one physical network may work, although I haven't tested it. What about a zone with 3 physical networks having traffic type of type guest. When I tried adding the 3rd Physical network with traffic type there was an error thrown that there is at least one physical network present in the zone having traffic type guest but not having a tag. I could add the guest traffic type to the third physical network only after adding tag to the second physical network as well. CC [@Pearl1594](https://github.com/Pearl1594) [@DaanHoogland](https://github.com/DaanHoogland)
> 
> Yes [@rajujith](https://github.com/rajujith) , there can be only one! (phys network for traffictype guest with no tag)

yes.

Assume there are two physical networks for Guest traffic (phy1 without tag, phy2 with tag xxx), when you add the third physical network for Guest traffic, you have to specify the tag (phy3 with tag yyy) . cc @rajujith

**Comment by rajujith** [comment]
After talking to @weizhouapache it is clear for me now. CloudStack expects exactly one physical network with a guest traffic type that is untagged. This is used for untagged isolated and L2 network offerings. For shared networks we can select the physical network while creating the network, although the same can be tagged as well. We need to test these scenarios for the https://github.com/apache/cloudstack/pull/10222 that fixes the listing issue for shared networks. 

To avoid the confusion we can do the following.
1. Improve the network offering listing for isolated network to display an error 'No physical network without tag available' 
2. Improve the documentation. I will do this myself. 

cc @DaanHoogland @Pearl1594

**Comment by rajujith** [comment]
@cmercado93 please check https://github.com/apache/cloudstack/issues/10316.

**Comment by DaanHoogland** [comment]
fixed by #10320 (#10222 )
